transport/src/main/java/io/camunda/zeebe/transport/stream/impl/ClientStreamManager.java

```diff
@@ -63,4 +62,19 @@ void remove(final UUID streamId) {
   void removeAll() {
     requestManager.removeAll(servers);
   }
+
+  public Boolean onPayloadReceived(final PushStreamRequest pushStreamRequest) {
+    final var streamId = pushStreamRequest.streamId();
+    final var payload = pushStreamRequest.payload();
+
+    final ClientStream<M> clientStream = registry.get(streamId);
+    if (clientStream != null) {
+      clientStream.getClientStreamConsumer().push(payload);
+      return true;
+    } else {
+      // client stream is closed
+      // TODO: remove stream from broker or return an error so that broker can close the stream
+      return false;
```

Comment:
:question: How to handle this case? 
- Return an error, Broker on receiving this error can remove the stream its registry.
- Client sends a explicit remove stream request